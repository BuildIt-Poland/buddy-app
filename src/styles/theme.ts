import { createMuiTheme, responsiveFontSizes } from '@material-ui/core/styles';
import { colorSchemes } from '@buildit/gravity-particles';

const { tealWhite } = colorSchemes;

let theme = createMuiTheme({
  typography: {
    htmlFontSize: 10, //62,5%
    button: {
      textTransform: 'none',
    },
    fontFamily:
      'Wipro Akkurat TT, Wipro Akkurat Mono TT, Roboto, Helvetica, Arial, sans-serif',
    h1: {
      fontSize: '3rem',
    },
    h2: {
      fontSize: '2.4rem',
    },
    h3: {
      fontSize: '2rem',
    },
    h4: {
      fontSize: '1.8rem',
    },
    h5: {
      fontSize: '1.6rem',
    },
    subtitle1: {
      fontSize: '1.4rem',
    },
    subtitle2: {
      fontSize: '1.2rem',
    },
  },
  palette: {
    type: 'light',
    primary: {
      main: tealWhite.groupB.accent,
      contrastText: tealWhite.groupA.neutral,
    },
    secondary: {
      main: tealWhite.groupB.neutral,
      contrastText: tealWhite.groupA.neutral,
    },
    background: {
      default: tealWhite.groupA.neutral,
    },
    text: {
      primary: tealWhite.groupB.neutral,
    },
  },
});

theme = responsiveFontSizes(theme);

export default theme;
