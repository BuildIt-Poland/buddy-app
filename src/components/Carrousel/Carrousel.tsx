import React from 'react';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import { makeStyles, Theme } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';

import { Newbie } from '../../../server/src/generated/schema-types';
import Avatar from '../Avatar';
import { ROUTES } from '../../shared/routes';

const useStyles = makeStyles(theme => ({
  carrouselElement: {
    height: '30rem',
    width: '20rem',
    padding: theme.spacing(2),
    borderRadius: '2rem',
    textDecoration: 'none',
  },
  carrousel: {
    [theme.breakpoints.down('sm')]: {
      overflowX: 'auto',
    },
  },
  link: {
    textDecoration: 'none',
  },
}));

const Carrousel: React.FC<{ newbies: Newbie[] }> = ({ newbies }) => {
  const classes = useStyles();
  const breakpointSm = useMediaQuery<Theme>(theme => theme.breakpoints.down('sm'));

  return (
    <Grid
      container
      spacing={2}
      wrap={breakpointSm ? 'nowrap' : 'wrap'}
      className={classes.carrousel}>
      {newbies.map(newbie => {
        const startDate = new Date(newbie.startDate);
        return (
          <Grid key={newbie.id} item>
            <Link
              className={classes.link}
              to={ROUTES.BUDDY_TASKS_LIST.replace(':newbieId', newbie.id)}>
              <Paper className={classes.carrouselElement}>
                <Avatar
                  progress={newbie.tasksInfo.buddyProgress * 100}
                  name={newbie.name}
                  position={newbie.position || undefined}></Avatar>
                {newbie.startDate && (
                  <Box m={2}>
                    <Typography align={'center'} component='p' variant='body2'>
                      <Box component='span' fontWeight={'bold'}>
                        Start Date
                      </Box>
                    </Typography>
                    <Typography align={'center'} component='p' variant='body2'>
                      {`${startDate.getDate()}-${startDate.getMonth()}-${startDate.getFullYear()}`}
                    </Typography>
                  </Box>
                )}
              </Paper>
            </Link>
          </Grid>
        );
      })}
    </Grid>
  );
};

export default Carrousel;
