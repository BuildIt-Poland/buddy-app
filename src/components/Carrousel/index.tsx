import Carrousel from './Carrousel';

export default Carrousel;
