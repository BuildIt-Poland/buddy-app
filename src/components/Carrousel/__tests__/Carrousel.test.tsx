import React from 'react';
import { create } from 'react-test-renderer';
import ThemeProvider from '@material-ui/styles/ThemeProvider';
import { MemoryRouter } from 'react-router';
import theme from 'styles/theme';
import Carrousel from '../Carrousel';

jest.mock('buddy-app-schema', () => {});
jest.mock('@material-ui/core/Grid', () => 'Grid');
jest.mock('@material-ui/core/Paper', () => 'Paper');
jest.mock('@material-ui/core/Box', () => 'Box');
jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('components/Avatar', () => 'Avatar');

describe('Component - Carrousel', () => {
  describe('renders correctly', () => {
    const newbies = [
      {
        id: 'ck17svulh9k2k0b17j31ansfk',
        photo: null,
        name: 'Test 2',
        startDate: null,
        tasksInfo: { buddyProgress: 0.5 },
      },
      {
        id: 'ck17swp2m9kcv0b17we0ibrdn',
        photo: null,
        name: 'Test 1',
        startDate: '2019-01-01',
        position: 'Front End',
        tasksInfo: { buddyProgress: 0.6 },
      },
    ];

    test('Should render correctly Carrousel', () => {
      const component = create(
        <MemoryRouter>
          <ThemeProvider theme={theme}>
            <Carrousel newbies={newbies} />
          </ThemeProvider>
        </MemoryRouter>
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
