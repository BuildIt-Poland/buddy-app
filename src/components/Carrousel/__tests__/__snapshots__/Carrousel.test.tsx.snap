// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Carrousel renders correctly Should render correctly Carrousel 1`] = `
<Grid
  className="makeStyles-carrousel-2"
  container={true}
  spacing={2}
  wrap="wrap"
>
  <Grid
    item={true}
  >
    <a
      className="makeStyles-link-3"
      href="/buddy/newbies/ck17svulh9k2k0b17j31ansfk/tasks"
      onClick={[Function]}
    >
      <Paper
        className="makeStyles-carrouselElement-1"
      >
        <Avatar
          imgSrc={null}
          name="Test 2"
          progress={50}
        />
      </Paper>
    </a>
  </Grid>
  <Grid
    item={true}
  >
    <a
      className="makeStyles-link-3"
      href="/buddy/newbies/ck17swp2m9kcv0b17we0ibrdn/tasks"
      onClick={[Function]}
    >
      <Paper
        className="makeStyles-carrouselElement-1"
      >
        <Avatar
          imgSrc={null}
          name="Test 1"
          progress={60}
          role="Front End"
        />
        <Box
          m={2}
        >
          <Typography
            align="center"
            component="p"
            variant="body2"
          >
            <Box
              component="span"
              fontWeight="bold"
            >
              Start Date
            </Box>
          </Typography>
          <Typography
            align="center"
            component="p"
            variant="body2"
          >
            2019-01-01
          </Typography>
        </Box>
      </Paper>
    </a>
  </Grid>
</Grid>
`;
